qd_proto/utils.lua

```diff
@@ -166,8 +166,7 @@ function utils.is_empty_str(str) return str == nil or str == '' end
 -- @param path Path to file.
 -- @return Filename.
 function utils.get_filename(path)
-    local start = path:find('[%w%s!-={-|]+[_%.].+')
```

Comment:
Please, see `utils.read_compact_long`. 
There is no handling of the case when `read_compact_int` returns `nil` (`compact_len > buf:len() - off`).

Also in code `read_compact_int`, `read_compact_long`:
`buf_len` is not the length of the buffer, but the length of the data that can be read, as far as I understand. 
Maybe you should rename it?